pyvista/core/_validation/_cast_array.py

```diff
@@ -122,7 +122,8 @@ def _cast_to_numpy(
     try:
         VisibleDeprecationWarning = np.exceptions.VisibleDeprecationWarning
     except AttributeError:
-        VisibleDeprecationWarning = np.VisibleDeprecationWarning
+        # we only type for newer numpy, and this branch only touched in older numpy
```

Comment:
As far as I can tell, we can only type based on a single version of numpy.  Should we declare the newest supported version of any dependency to be the version of truth here?

This is a particular problem supporting such wide ranges of dependency versions.  We could remove this pain if we narrow support to less versions of vtk and numpy.  Something to consider to avoid this maintenance headache.

Reply:
If I understand the issue it might also be an option to use `if not typing.TYPE_CHECKING: ...`.